Scala RPC Server fully tested with TS RPC Client

@Shanedell has reported that the search function is returning no matches when matches are expected.  

I think the best way forward is to run the Scala server with the RPC TC client tests as we do with the C++ server.  Search is being exercised by the TS client tests.  We need to make sure it's working there, then we have a good chance of getting it working on the front-end layer.

The tests will be a combination of `sbt run` to run the Scala server, then `npm test` to run the TS client tests.  I can script that in `build.sh`.

@arosien  On v0.9.14 I am also getting this error when trying to create a viewport

```bash
gRPC server bound to: /127.0.0.1:9000
[ERROR] [07/28/2022 20:49:14.846] [omega-grpc-akka.actor.default-dispatcher-4] [akka://omega-grpc/user/$a/L1VzZXJzL3NkZWxsL3dvcmtzcGFjZXMvZGFmZm9kaWwvc2FtcGxlV29ya3NwYWNlLy52c2NvZGUvbGF1bmNoLmpzb24=] 1024 is not a member of ValueEnum (0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512)
java.util.NoSuchElementException: 1024 is not a member of ValueEnum (0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512)
        at enumeratum.values.ValueEnum.$anonfun$withValue$1(ValueEnum.scala:58)
        at scala.Option.getOrElse(Option.scala:201)
        at enumeratum.values.ValueEnum.withValue(ValueEnum.scala:58)
        at enumeratum.values.ValueEnum.withValue$(ValueEnum.scala:57)
        at com.ctc.omega_edit.api.SessionEvent$.withValue(SessionEvent.scala:25)
        at com.ctc.omega_edit.api.SessionCallback.invoke(SessionCallback.scala:40)
        at com.ctc.omega_edit.api.SessionCallback.invoke$(SessionCallback.scala:30)
        at com.ctc.omega_edit.api.SessionCallback$$anonfun$apply$2.invoke(SessionCallback.scala:65)
        at jnr.ffi.provider.jffi.NativeClosureProxy$$impl$$1.invoke(Unknown Source)
        at com.kenai.jffi.Foreign.invokeArrayReturnLong(Native Method)
        at com.kenai.jffi.Invoker.invokeLong(Invoker.java:1970)
        at com.kenai.jffi.Invoker$LP64.invokeAddress(Invoker.java:2223)
        at com.ctc.omega_edit.FFI$jnr$ffi$0.omega_edit_create_viewport(Unknown Source)
        at com.ctc.omega_edit.SessionImpl.viewCb(SessionImpl.scala:116)
        at com.ctc.omega_edit.grpc.Session$$anonfun$receive$1.applyOrElse(Session.scala:123)
        at akka.actor.Actor.aroundReceive(Actor.scala:537)
        at akka.actor.Actor.aroundReceive$(Actor.scala:535)
        at com.ctc.omega_edit.grpc.Session.aroundReceive(Session.scala:98)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:577)
        at akka.actor.ActorCell.invoke(ActorCell.scala:547)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
        at akka.dispatch.Mailbox.run(Mailbox.scala:231)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

[ERROR] [07/28/2022 20:49:15.844] [omega-grpc-akka.actor.default-dispatcher-10] [akka.grpc.scaladsl.GrpcExceptionHandler(akka://omega-grpc)] Unhandled error: [Ask timed out on [Actor[akka://omega-grpc/user/$a#-211926991]] after [1000 ms]. Message of type [com.ctc.omega_edit.grpc.Editors$SessionOp]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.] (akka.pattern.AskTimeoutException: Ask timed out on [Actor[akka://omega-grpc/user/$a#-211926991]] after [1000 ms]. Message of type [com.ctc.omega_edit.grpc.Editors$SessionOp]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply.)
```

--- a/build.sh
+++ b/build.sh
@@ -77,7 +77,9 @@ kill $server_pid
 
 if [ $test_scala_server -ne 0 ]; then
   pushd src/rpc/server/scala
-  sbt run
+  sbt stage
+  ./target/universal/stage/bin/omega-edit-grpc-server --port 9000 &
+  server_pid=$!
   popd
   pushd src/rpc/client/ts/
   npm install
@@ -85,7 +87,7 @@ if [ $test_scala_server -ne 0 ]; then
   npm run lint
   npm test
   popd
-  # TODO: kill the Scala RPC server
+  kill $server_pid
 fi
 
 rm -rf build-tests-integration-$type

--- a/src/rpc/server/scala/build.sbt
+++ b/src/rpc/server/scala/build.sbt
@@ -22,7 +22,7 @@ lazy val packageData = Json
   .as[JsObject]
 lazy val omegaVersion = packageData("version").as[String]
 
-name := "example-grpc-server"
+name := "omega-edit-grpc-server"
 scalaVersion := "2.13.8"
 
 lazy val ghb_repo_owner = "ctc-oss"
@@ -46,9 +46,12 @@ startYear := Some(2021)
 publishConfiguration := publishConfiguration.value.withOverwrite(true)
 publishLocalConfiguration := publishLocalConfiguration.value.withOverwrite(true)
 
+fork := true
+
 libraryDependencies ++= Seq(
   "com.ctc" %% "omega-edit" % omegaVersion,
   "com.ctc" %% "omega-edit-native" % omegaVersion classifier s"${arch.id}",
+  "com.monovore" %% "decline" % "2.3.0",
   "org.scalatest" %% "scalatest" % "3.2.12" % Test
 )
 
@@ -64,4 +67,4 @@ externalResolvers ++= Seq(
 
 Compile / PB.protoSources += baseDirectory.value / "../../protos"
 
-enablePlugins(AkkaGrpcPlugin, GitVersioning, JavaAppPackaging, UniversalPlugin)
+enablePlugins(AkkaGrpcPlugin, GitVersioning, JavaServerAppPackaging, UniversalPlugin)

--- a/src/rpc/server/scala/src/main/scala/com/ctc/omega_edit/grpc/boot.scala
+++ b/src/rpc/server/scala/src/main/scala/com/ctc/omega_edit/grpc/boot.scala
@@ -17,14 +17,39 @@
 package com.ctc.omega_edit.grpc
 
 import akka.actor.ActorSystem
+import com.monovore.decline._
 
 import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.concurrent.Await
 
-object boot extends App {
-  implicit val sys: ActorSystem = ActorSystem("omega-grpc")
+object boot
+    extends CommandApp(
+      name = "omega-edit-grpc-server",
+      header = "",
+      main = Opts
+        .option[Int]("port", short = "p", metavar = "port_num", help = "Set the gRPC port to listen on. Default: 9000")
+        .withDefault(9000)
+        .map(new boot(_).run())
+    )
+
+class boot(port: Int) {
+  implicit val sys: ActorSystem = ActorSystem("omega-grpc-server")
   implicit val ec: ExecutionContext = sys.dispatcher
 
-  EditorService.bind().foreach { binding =>
-    println(s"gRPC server bound to: ${binding.localAddress}")
+  def run() = {
+    val done =
+      for {
+        binding <- EditorService.bind(port = port)
+
+        _ = println(s"gRPC server bound to: ${binding.localAddress}")
+
+        done <- binding.addToCoordinatedShutdown(1.second).whenTerminated
+
+        _ = println(s"exiting...")
+      } yield done
+
+    Await.result(done, atMost = Duration.Inf)
+    ()
   }
 }

